src/jaxsim/rbda/contacts/common.py

```diff
@@ -14,6 +18,47 @@
     from typing_extensions import Self
 
 
+@functools.partial(jax.jit, static_argnames=("terrain",))
+def compute_penetration_data(
+    p: jtp.VectorLike,
+    v: jtp.VectorLike,
+    terrain: jaxsim.terrain.Terrain,
+) -> tuple[jtp.Float, jtp.Float, jtp.Vector]:
+    """
+    Compute the penetration data (depth, rate, and terrain normal) of a collidable point.
+
+    Args:
+        p: The position of the collidable point.
+        v:
+            The linear velocity of the point (linear component of the mixed 6D velocity
+            of the implicit frame `C = (W_p_C, [W])` associated to the point).
+        terrain: The considered terrain.
+
+    Returns:
+        A tuple containing the penetration depth, the penetration velocity,
+        and the considered terrain normal.
+    """
+
+    # Pre-process the position and the linear velocity of the collidable point.
+    W_ṗ_C = jnp.array(v).squeeze()
+    px, py, pz = jnp.array(p).squeeze()
+
+    # Compute the terrain normal and the contact depth.
+    n̂ = terrain.normal(x=px, y=py).squeeze()
+    h = jnp.array([0, 0, terrain.height(x=px, y=py) - pz])
+
+    # Compute the penetration depth normal to the terrain.
+    δ = jnp.maximum(0.0, jnp.dot(h, n̂))
+
+    # Compute the penetration normal velocity.
+    δ̇ = -jnp.dot(W_ṗ_C, n̂)
+
+    # Enforce the penetration rate to be zero when the penetration depth is zero.
+    δ̇ = jnp.where(δ > 0, δ̇, 0.0)
+
+    return δ, δ̇, n̂
```

Comment:
I guess this is a Unicode problem of the accent not handled correctly, but this is a bit confusing, as the dot goes in the subsequent comma at least on my machine:

![unicodeaccent](https://github.com/user-attachments/assets/4af7947e-9cb6-41c9-a647-60012a9d7763)

Reply:
We checked with @xela-95, and the problem is not there in Chrome, but it is in Visual Studio Code:

![image](https://github.com/user-attachments/assets/466b057d-9aea-45ee-b5a1-93c80f90ae95)

perhaps we can try to avoid to use this character until it is supported in Visual Studio Code? I guess not a lot of people use Firefox, but Visual Studio Code instead is quite used.

Reply:
Generally speaking, I usually never use unicode characters in public APIs like function arguments. This would mean that if the user wants to pass in their downstream code a `kwarg`, they have to know how to deal with unicode. This is something to avoid.

The situation is different for logic in function body. Here, I value more having code that is 1) readable and 2) as close as possible to pseudocode. Dotted variables are use pretty much everywhere in JaxSim, I'm curious, why this case is different than all other ones?

Reply:
I totally like dotted variables (when they work). My guess is that the difference here is that diacritics with greek letters are not supported uniformly across fonts.

Reply:
Ok now I understood your point. I agree that in few cases, dotted greek letters look strange. Addressed in https://github.com/ami-iit/jaxsim/compare/2b6f69d4755d37dbf5972c074d6eda5f8951ecb1..85a98dabe7afa88ef9b7c913d4b8875e2346e244, thanks!

Reply:
Great, thanks! Perhaps in some time the default fonts in Visual Studio Code will address the problem and we can start using it again.

Reply:
For the records, my Python IDE is PyCharm with the **Fira Code** font (that supports ligatures, which are beautiful), and dotted greek letters look as follows:

![image](https://github.com/user-attachments/assets/f9cbcbec-c88b-4b16-8dfe-e0280666d667)

Reply:
Yes, also on Google Chrome it works fine, it is probably an issue of default fonts.